File: app/src/main/java/yangFenTuoZi/runner/plus/utils/ThemeUtils.java
package yangFenTuoZi.runner.plus.utils;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.StyleRes;

import rikka.core.util.ResourceUtils;
import yangFenTuoZi.runner.plus.R;
import yangFenTuoZi.runner.plus.info.Info;

public class ThemeUtils {

    @StyleRes
    public static int getTheme(Context context) {
        return getTheme(isDark(context));
    }

    @StyleRes
    public static int getTheme(boolean isDark) {
        return isDark ? R.style.Theme : R.style.Theme_Light;
    }

    public static boolean isDark(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(Info.APPLICATION_ID + "_preferences", 0);
        String dark_theme = sharedPreferences.getString("dark_theme", "MODE_NIGHT_FOLLOW_SYSTEM");
        return switch (dark_theme) {
            case "MODE_NIGHT_FOLLOW_SYSTEM" ->
                    ResourceUtils.isNightMode(context.getResources().getConfiguration());
            case "MODE_NIGHT_YES" -> true;
            default -> false;
        };
    }
}

File: app/src/main/java/yangFenTuoZi/runner/plus/utils/ThemeUtils.kt
package yangFenTuoZi.runner.plus.utils

import android.content.Context
import androidx.annotation.StyleRes
import rikka.core.util.ResourceUtils
import yangFenTuoZi.runner.plus.BuildConfig
import yangFenTuoZi.runner.plus.R

object ThemeUtils {
    @StyleRes
    fun getTheme(context: Context): Int {
        return getTheme(isDark(context))
    }

    @StyleRes
    fun getTheme(isDark: Boolean): Int {
        return if (isDark) R.style.Theme else R.style.Theme_Light
    }

    fun isDark(context: Context): Boolean {
        val sharedPreferences =
            context.getSharedPreferences(BuildConfig.APPLICATION_ID + "_preferences", 0)
        val darkTheme: String =
            sharedPreferences.getString("dark_theme", "MODE_NIGHT_FOLLOW_SYSTEM")!!
        return when (darkTheme) {
            "MODE_NIGHT_FOLLOW_SYSTEM" -> ResourceUtils.isNightMode(
                context.resources.configuration
            )

            "MODE_NIGHT_YES" -> true
            else -> false
        }
    }
}